src/components/DropzoneAreaFunc.js

```diff
@@ -0,0 +1,486 @@
+
+const styles = {
+  "@keyframes progress": {
+    "0%": {
+      backgroundPosition: "0 0",
+    },
+    "100%": {
+      backgroundPosition: "-70px 0",
+    },
+  },
+  dropZone: {
+    position: "relative",
+    width: "100%",
+    minHeight: "250px",
+    backgroundColor: "#F0F0F0",
+    border: "dashed",
+    borderColor: "#C8C8C8",
+    cursor: "pointer",
+    boxSizing: "border-box",
+  },
+  stripes: {
+    border: "solid",
+    backgroundImage:
+      "repeating-linear-gradient(-45deg, #F0F0F0, #F0F0F0 25px, #C8C8C8 25px, #C8C8C8 50px)",
+    animation: "progress 2s linear infinite !important",
+    backgroundSize: "150% 100%",
+  },
+  rejectStripes: {
+    backgroundImage:
+      "repeating-linear-gradient(-45deg, #fc8785, #fc8785 25px, #f4231f 25px, #f4231f 50px)",
+  },
+  dropzoneTextStyle: {
+    textAlign: "center",
+  },
+  uploadIconSize: {
+    width: 51,
+    height: 51,
+    color: "#909090",
+  },
+  dropzoneParagraph: {
+    marginBottom: 20,
+    marginTop: 20,
+  },
+};
+
+const snackbarAnchorOrigin = {
+  vertical: "bottom",
+  horizontal: "left",
+};
+
+/**
+ * This components creates a Material-UI Dropzone, with previews and snackbar notifications.
+ */
+const DropzoneAreaFunc = (props) => {
+  const {
+    acceptedFiles,
+    classes,
+    clearOnUnmount,
+    disableRejectionFeedback,
+    dropzoneClass,
+    dropzoneParagraphClass,
+    dropzoneText,
+    filesLimit,
+    getDropRejectMessage,
+    getFileAddedMessage,
+    getFileLimitExceedMessage,
+    getFileRemovedMessage,
+    initialFiles,
+    maxFileSize,
+    onChange,
+    onDelete,
+    onDrop,
+    onDropRejected,
+    previewChipProps,
+    previewGridClasses,
+    previewGridProps,
+    previewText,
+    showAlerts,
+    showFileNames,
+    showFileNamesInPreview,
+    showPreviews,
+    showPreviewsInDropzone,
+    useChipsForPreview,
+  } = props;
+
+  const [fileObjects, setFileObjects] = useState([]);
+  const [openSnackBar, setOpenSnackBar] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState("");
+  const [snackbarVariant, setSnackbarVariant] = useState("success");
+
+  useEffect(() => {
+    filesArray(initialFiles);
+
+    return () => {
+      if (clearOnUnmount) {
+        setFileObjects([]);
+
+        if (onChange) {
+          onChange([]);
+        }
+      }
+    };
+  }, []);
+
+  const filesArray = async (urls) => {
+    try {
+      const fileObjs = await Promise.all(
+        urls.map(async (url) => {
+          const file = await createFileFromUrl(url);
+          const data = await readFile(file);
+
+          return {
+            file,
+            data,
+          };
+        })
+      );
+
+      setFileObjects([].concat(fileObjects, fileObjs));
+      if (onChange) {
+        onChange(fileObjects.map((fileObject) => fileObject.file));
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  const handleSnackbarState = (shouldOpen, message, variant) => {
```

Comment:
See comment above related to `Snackbar state management`.

Should become the `showSnackbar` callback, move it inside the custom hook and wrap it inside a `useCallback` hook.